spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'spec_helper'
+
+describe SessionsController do
+  describe "GET new" do
+    it "renders the new template for unauthenticated users" do
+      get :new
+      expect(response).to render_template :new
+    end
+    it "redirects to the home page for authenticated users" do
+      session[:user_id] = Fabricate(:user).id
+      get :new
+      expect(response).to redirect_to home_path
+    end
+  end
+
+  describe "POST create" do
+    context "with valid credentials" do
+      let (:alice) {Fabricate(:user)}
+      it "puts the signed in user in the session" do
+        post :create, email: alice.email, password: alice.password
+        expect(session[:user_id]).to eq(alice.id)
+      end
+
+      it "redirects to the home path" do 
+        post :create, email: alice.email, password: alice.password
+        expect(response).to redirect_to home_path
+      end
+
+      it "sets the notice" do
+        post :create, email: alice.email, password: alice.password
+        expect(flash[:notice]).not_to be_blank
+      end
+    end
+
+    context "with invalid credentials" do
+      let(:alice) {Fabricate(:user)}
+      it "doesn't put the user in the session" do
+        post :create, email: alice.email, password: alice.password + "wrong_password"
```

Comment:
`password: alice.password + "wrong_password"` - I think it would be easier to just type wrong password straight rather than having `alice.password` + something.